src/playlist-maker.js

```diff
@@ -1,7 +1,57 @@
 import videojs from 'video.js';
+import * as guid from './guid';
 import playItem from './play-item';
 import * as autoadvance from './auto-advance';
 
+/**
+ * Generate a unique id for each playlist item object. This id will be used to determine
+ * index of an item in the playlist array for cases where there are multiple items with
+ * the same source set.
+ *
+ * @private
+ *
+ * @param {Array}
+ *         An array of playlist items
+ *
+ * @return {Array}
+ *          The list of playlist items with unique ids
+ */
+const generatePlaylistItemId = (arr) => {
+  const list = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const id = guid.newGUID();
+
+    if (arr[i] && typeof arr[i] === 'object') {
```

Comment:
Do we need to check this? I think this whole function could be reduced to:

```js
arr.forEach(item => {
  item.playlistItemId_ = newGUID();
});
```

Reply:
If we have non-object values in the array, that's an error and we should fail.

Reply:
Yep, that looks much better! W.r.t non-object values in the list, I'm thinking perhaps it would be best to what @gkatsev mentioned, ie normalise list items that are passed to the plugin by making sure they are objects.